### PR TITLE
adds a mednist 2D registration demo

### LINKED ISSUE #5440: adds a 2d registration bundle
**Is your feature request related to a problem? Please describe.**

in addition to https://github.com/Project-MONAI/model-zoo/issues/171, a learning based registration bundle could be created based on https://github.com/Project-MONAI/tutorials/tree/main/2d_registration

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes https://github.com/Project-MONAI/MONAI/issues/5440

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [x] In-line docstrings updated.
- [x] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [x] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [x] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [x] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [x] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [x] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

File: models/mednist_reg/configs/metadata.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    "version": "0.0.1",
+    "changelog": {
+        "0.0.1": "Initial version"
+    },
+    "monai_version": "1.1.0",
+    "pytorch_version": "1.13.0a0+d0d6b1f",
```
Comment (yiheng-wang-nv) [inline models/mednist_reg/configs/metadata.json]: Thanks @wyli . Here could the pytorch version be a formal version, like `1.13.0`?

Comment (wyli) [reply models/mednist_reg/configs/metadata.json]: this is autogenerated by https://github.com/Project-MONAI/MONAI/blob/5e6f105ff1d40266debfbd9774a9a0bdbe4d817a/monai/bundle/utils.py#L35

do you think we should add a split there?
```
>>> "1.13.0a0+d0d6b1f".split('a')[0]
'1.13.0'
```
we'll always have this kind of pytorch versioning in our docker image.

File: models/mednist_reg/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+Training from an existing checkpoint file, for example, `models/model_key_metric=-0.0734.pt`:
+```bash
+python -m monai.bundle run training --config_file configs/train.yaml [...omitting other args] --ckpt "models/model_key_metric=-0.0734.pt"
+```
+
+
+## Inference
+
+The following figure shows an intra-subject (`--cross_subjects false`) model inference results (Fixed, moving and predicted images from left to right)
+
```
Comment (yiheng-wang-nv) [inline models/mednist_reg/docs/README.md:40]: Here could "cuda:1" be emphasized like other examples below and above (for `5` epochs, for example, `models/model_key_metric=-0.0734.pt`) ?

File: models/mednist_reg/configs/inference.yaml [hunk 3]
```diff
@@ -0,0 +1,148 @@
+---
+imports:
+  - $import glob
+  - $import matplotlib.pyplot as plt
+dataset_dir: "../MedNIST/Hand"
+datalist: "$list(sorted(glob.glob(@dataset_dir + '/*.jpeg')))[8500:8510]"
```
Comment (yiheng-wang-nv) [inline models/mednist_reg/configs/inference.yaml:7]: suggest to add a comment like in `train.yaml`
```
# inference with 10 images
```

File: models/mednist_reg/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+```bash
+python -m monai.bundle run display --config_file configs/train.yaml
+```
+
+```bash
+python -m monai.bundle run display --config_file configs/train.yaml --cross_subjects true
+```
+
+
```
Comment (yiheng-wang-nv) [inline models/mednist_reg/docs/README.md]: Here could the "extra" blank lines be removed?

**Review by yiheng-wang-nv — APPROVED** [review]

**Comment by yiheng-wang-nv** [comment]
Thanks @wyli for contributing this bundle, the PR looks good to me, a few comments are added above and I do not have other questions.

**Pull request merged**